### PR TITLE
[kube-prometheus-stack] - add scrapeConfig CR and tpl support for all CR's selectors

### PR DESCRIPTION
**Pull request by Allex1**
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by Allex1** [comment]
@jkroepke @monotek 🙏

File: charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml [hunk 1]
```diff
@@ -65,7 +65,7 @@ spec:
 {{- end }}
 {{- if .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector }}
   alertmanagerConfigNamespaceSelector:
-{{ toYaml .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector | indent 4}}
+{{ tpl (toYaml .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector | indent 4) . }}
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml:68]: I do not have any merge permission for kube-prometheus-stack. However, can you how me the use-case here?

Comment (Allex1) [reply charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml:68]: > I do not have any merge permission for kube-prometheus-stack.

darn it 😊
> However, can you how me the use-case here?

It allows you to use helm global variables like `{{.Release.Namespace}}` in values.yaml for those specific keys.

https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function

**Comment by jkroepke** [comment]
For completeness, can you add

* spec.scrapeConfigNamespaceSelector
* spec.scrapeConfigSelector

as well? They are part of the lastest update, but not cover yet.

File: charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml (hunk 1 above)
Comment (jkroepke) [reply charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml:68]: I can fully understand the reason, but using the `tpl (toYaml .) $` always has some limitation. For example, the pattern only works exactly to one namespace, like:

```yaml
    probeNamespaceSelector:
      matchLabels:
        kubernetes.io/name: {{ $.Release.Namespace }}
```

or

```yaml
    probeNamespaceSelector:
      matchExpressions:
        - key: name
          operator: in
          values:
            - {{ $.Release.Namespace }}
```

However, having a list of namespaces would not work here.

```mustache
    probeNamespaceSelector:
      matchExpressions:
        - key: name
          operator: in
          values: ["{{ join "\",\"" $.Values.namespaces}}"]
```

To resolve the limitation (for example on the [Alertmanager config](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/alertmanager/secret.yaml#L17)), a pure string can be passed to the tpl function.

But I don't if we want this here. I would just like to share my insights with the `tpl (toYaml .) $` pattern.

Comment (Allex1) [reply charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml:68]: > having a list of namespaces would not work here

You can test this assumption with my pr

**Comment by Allex1** [comment]
> For completeness, can you add
> 
> * spec.scrapeConfigNamespaceSelector
> * spec.scrapeConfigSelector
> 
> as well? They are part of the lastest update, but not cover yet.

done

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 46.4.2
+version: 46.4.3
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/Chart.yaml:24]:
```suggestion
version: 46.5.0
```

**Review by QuentinBisson — APPROVED** [review]

**Comment by onedr0p** [comment]
I don't see where teh scrapeConfig CR was added in this PR?

If you don't have it installed this update will fail:

```
Prometheus/monitoring/kube-prometheus-stack dry-run failed: failed to create typed patch object (monitoring/kube-prometheus-stack; monitoring.coreos.com/v1, Kind=Prometheus): .spec.scrapeConfigNamespaceSelector: field not declared in schema
```

**Pull request merged**